### DISCUSSION TITLE
Renaming of the cell when using import_gds and gf.cell decorator

### DISCUSSION THREAD

**Discussion by Paudet-ul** [gdsfactory/gdsfactory#2225] [Q&A] [answered]
Hi,

I'm currently facing an issue regarding the name of the cell after importing it via import_gds. My case is really similar to the behavior observed on the ubc pdk.

For example, when importing a fixed cell as a component like:
```
@gf.cell
def ebeam_adiabatic_te1550() -> gf.Component:
    """Return ebeam_adiabatic_te1550 fixed cell."""
    return import_gds("ebeam_adiabatic_te1550.gds")
```
The cell is first named 'ebeam_adiabatic_te1550' after being imported with import_gds. But, when the component is returned by the gf.cell decorator, the name is changed to "ebeam_adiabatic_te1550$1". In my case, I need to have the original cell name, without any altercation ($1). 

I tried,
```
c = components.ebeam_adiabatic_te1550(autoname=False)
c.name = 'ebeam_adiabatic_te1550'
```
but it increases the name to 'ebeam_adiabatic_te1550$1' automatically. It has something to do with the cache, but I'm not sure how to fix this.

I've looked at https://github.com/gdsfactory/gdsfactory/discussions/1547. With the proposed solution, I end up with a nested cell named 'ebeam_adiabatic_te1550$1' containing the cell 'ebeam_adiabatic_te1550'. I would like to have a single cell named 'ebeam_adiabatic_te1550', but every time I'm trying to use this name, it adds a $1 to it.

Any help would be highly appreciated.

**Comment by Paudet-ul** [comment]
I did a bit of digging, and it looks like import_gds adds something in the cache (import_gds_7c7a70c6). This component points to the reference (ebeam_adiabatic_te1550). Therefore, I think that when the decorator is executed over the component, it finds that the name (ebeam_adiabatic_te1550) is already under another component, which might be why the $1 is added in the name. The full cache after calling `components.ebeam_adiabatic_te1550` is:
```
{'import_gds_7c7a70c6':  ebeam_adiabatic_te1550: uid bbaf093d, ports ['o2', 'o1', 'o3', 'o4'], references ['Adiabatic3dB_TE_FullEtc h_1'], 1 polygons, 
'ebeam_adiabatic_te1550': ebeam_adiabatic_te1550$1: uid 4e3e4167, ports ['o2', 'o1', 'o3', 'o4'], references ['Adiabatic3dB_TE_FullEtch_1'], 1 polygons} 
```

Is there a way to prevent import_gds_uid to be added to the cache? I don't see the use for that.

**Comment by joamatab** [comment, marked as answer]
have you tried?

```
@gf.functions.cache
def ebeam_adiabatic_te1550() -> gf.Component:
    """Return ebeam_adiabatic_te1550 fixed cell."""
    return import_gds("ebeam_adiabatic_te1550.gds")

```

**Reply by tvt173** [reply]
can't you also just do 

```python
@gf.cell
def ebeam_adiabatic_te1550() -> gf.Component:
    """Return ebeam_adiabatic_te1550 fixed cell."""
    return import_gds("ebeam_adiabatic_te1550.gds", cache=False)
```

?

**Reply by Paudet-ul** [reply]
I tried using cache=False argument in the import_gds call, but it still adds the import_gds component to the cache as well as the cell. It doesn't seem to affect the cache.

The gf.functions.cache decorator works! It seems to do the same thing as if I'm simply not using any decorator over the function. How is it different than using the gf.cell decorator?

**Reply by tvt173** [reply]
`gf.functions.cache` is simply a wrapper for [functools.lru_cache](https://docs.python.org/3/library/functools.html#functools.lru_cache), which generically caches the output of _any_ function. the `@cell` decorator is a special decorator just for cells. the way it does caching is a bit different, and it also does some special things like locking the cell and giving it a unique name. in your case, it's not necessary though, because you're just re-returning a component which has already been created with `import_gds`, which already uses the `@cell` wrapper (and, importantly, it already has and reserves the correct cell name that you desire)

i'd say if the above solution with `cache=False` doesn't work, that's a bug, don't you think @joamatab ?
